src/WooCommerce.php

```diff
@@ -223,7 +223,14 @@ public static function woocommerce_process_product_meta($post_id) {
     foreach ($custom_fields as $field) {
       if (isset($_POST[$field])) {
         if (!is_array($_POST[$field]) && $_POST[$field]) {
-          update_post_meta($post_id, $field, $_POST[$field]);
+          if ($field !== '_' . Plugin::PREFIX . '_gtin') {
+            update_post_meta($post_id, $field, $_POST[$field]);
+          } else if (!self::is_existing_gtin($post_id, $_POST[$field])) {
+            // GTIN is only saved if unique.
+            update_post_meta($post_id, $field, $_POST[$field]);
+          } else {
+            \WC_Admin_Meta_Boxes::add_error(__('The entered GTIN isn\'t unique and couldn\'t be saved. Please enter a unique GTIN and save again. All other changes have been saved.', Plugin::L10N));
+          }
```

Comment:
I'd propose to refactoring the conditional logic to ensure the error message is displayed only in case the field is GTIN and the GTIN is existing.

Something like:
```
if ($field !== '_' . Plugin::PREFIX . '_gtin') {
  update_post_meta($post_id, $field, $_POST[$field]);
}
else {
  if (!self::is_existing_gtin($post_id, $_POST[$field])) {
    // update
  }
  else {
    // error
  }
}
```